roles/ceph-rgw/tasks/rgw_create_pools.yml

```diff
@@ -0,0 +1,74 @@
+---
+  - name: remove ec profile
+    command: "{{ container_exec_cmd }} ceph --connect-timeout 5 --cluster {{ cluster }} osd erasure-code-profile rm {{ item.value.ec_profile }}"
+    with_dict: "{{ rgw_create_pools }}"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    changed_when: false
+    run_once: true
+    register: result
+    when: 
+      - item.value.pool_type is defined  
+      - item.value.pool_type == 'ec'
+    ignore_errors: yes
+
+  - name: set ec profile
+    command: "{{ container_exec_cmd }} ceph --connect-timeout 5 --cluster {{ cluster }} osd erasure-code-profile set {{ item.value.ec_profile }} k={{ item.value.ec_k }} m={{ item.value.ec_m }}"
+    with_dict: "{{ rgw_create_pools }}"
+    delegate_to: "{{ groups[mon_group_name][0] }}"
+    changed_when: false
+    run_once: true
+    register: result
+    when: 
+      - item.value.pool_type is defined  
+      - item.value.pool_type == 'ec'
```

Comment:
@guits should I check to make sure `ec_k` and `ec_m` are defined before this job is run?

Reply:
> @guits should I check to make sure `ec_k` and `ec_m` are defined before this job is run?

That wouldn't hurt :slightly_smiling_face: 

> Also is there a way to fail the job if those variables are not defined but `pool_type` is set to `ec`?

like an early check in ceph-validate?

Reply:
Done. Take a look at the changes I just made to the ceph-validate task.